creusot/src/translation/function/place.rs

```diff
@@ -180,11 +180,18 @@ pub fn translate_rplace_inner<'tcx>(
                 }
             }
             Field(ix, _) => match place_ty.ty.kind() {
-                TyKind::Adt(def, _) => {
+                TyKind::Adt(def, subst) => {
+                    use rustc_middle::ty::DefIdTree;
                     let variant_id = place_ty.variant_index.unwrap_or_else(|| 0u32.into());
                     let variant = &def.variants()[variant_id];
 
-                    ctx.translate_accessor(def.variants()[variant_id].fields[ix.as_usize()].did);
+                    let field_id = variant.fields[ix.as_usize()].did;
+                    // Filter out enum variants, and maybe other things...
+                    if !matches!(util::item_type(ctx.tcx, variant.def_id), ItemType::Unsupported(_))
+                    {
+                        names.insert(ctx.tcx.parent(field_id), subst);
+                    }
```

Comment:
this can just be replaced by `names.insert(def.did(), subst)`, which will correctly add the edge for the datatype itself.
```suggestion
                    
                    names.insert(def.did(), subst);
                    
```